src/librustc_errors/emitter.rs

```diff
@@ -383,6 +346,52 @@ impl EmitterWriter {
         }
         output
     }
+}
+
+impl EmitterWriter {
```

Comment:
keep this impl block above `impl FileWithAnnotatedLines {` to reduce diff size

Reply:
Just squashed the commits and moved the impl block. The diff got bigger, but I think it's easier to understand now.

Reply:
oh heh, I should have maybe expanded that source. I was hoping for just some changs in `}` and signatures, oh well. lgtm now